src/mtype.d

```diff
@@ -9497,27 +9453,69 @@ public:
         return params;
     }
 
+    extern (D):
```

Comment:
> Because the _foreach function being D does not make the delegate parameter D.

You need to add `extern (D)` to the delegate type alias `ForeachDg`, e.g.

`alias ForeachDg = extern(D) int delegate(size_t paramidx, Parameter param);`

> If there's a mismatch in a .h file, and somebody actually uses it, there'll be a linker error. So I'm not concerned about it until/unless that happens.

I dislike such the rough change. It's bug-prone method.